wallet-new/src/Cardano/Wallet/WalletLayer/Types.hs

```diff
@@ -288,6 +296,48 @@ instance Buildable UpdateAccountError where
     build (UpdateAccountWalletIdDecodingFailed txt) =
         bprint ("UpdateAccountWalletIdDecodingFailed " % build) txt
 
+------------------------------------------------------------
+-- Errors when getting Transactions
+------------------------------------------------------------
+
+data GetTxError =
+      GetTxMissingWalletIdError
+    | GetTxAddressDecodingFailed Text
+    | GetTxInvalidSortingOperaration String
+    -- throwM MissingRequiredParams { requiredParams = pure ("wallet_id", "WalletId") }
+
+instance Show GetTxError where
+    show = formatToString build
+
+instance Buildable GetTxError where
+    build GetTxMissingWalletIdError =
+        bprint "GetTxMissingWalletIdError "
+    build (GetTxAddressDecodingFailed txt) =
+        bprint ("GetTxAddressDecodingFailed " % build) txt
+    build (GetTxInvalidSortingOperaration txt) =
+        bprint ("GetTxInvalidSortingOperaration " % build) txt
+
+
+instance Arbitrary GetTxError where
+    arbitrary = oneof [ pure GetTxMissingWalletIdError
+                      , pure (GetTxAddressDecodingFailed "by_amount")
+                      , pure (GetTxInvalidSortingOperaration "123")
+                      ]
+
+instance Exception GetTxError
+
+------------------------------------------------------------
+-- General-purpose errors which may arise when working with
+-- the wallet layer
+------------------------------------------------------------
+
+data WalletLayerError =
+    InvalidAddressConversionFailed Text
+    -- ^ Trying to decode the input 'Text' into a Cardano 'Address' failed
+    deriving Show
```

Comment:
This `WalletLayerError` is actually completely gone from `develop` after some of my refactoring. Do you think it would be possible to avoid reintroducing it?